training/level-2-command-line-interfaces/dragon-warrior/tmarsha1/file_ops.py

```diff
@@ -1,21 +1,125 @@
-"""This module provides various functions for operating on files"""
+"""
+This module provides various functions for operating on files
+
+Functions:
+    copy_files: copy file(s) to a specific location.
+    move_files: move file(s) to a specific location.
+"""
 
 import subprocess
 
 
-def copy_files(files: list, destination: str):
+class FolderNotFoundError(Exception):
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
+        return repr(self.value)
+        
+
+class InsufficientRightsError(Exception):
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
+        return repr(self.value)
+
+
+def _process_files(
+        command: str, 
+        options: str, 
+        file: str, 
+        destination: str
+) -> str:
+    try:
+        byte_result = subprocess.check_output(
+            args=[command, options, file, destination],
+            stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as file_error:
+        message = file_error.output.decode()
+
+        if "cannot stat" in message:
+            raise FileNotFoundError()
+        elif "Not a directory" in message:
+            raise FolderNotFoundError(destination)
+        elif "Permission" in message:
+            raise InsufficientRightsError(destination)
+        else:
+            print(message)
+    else:
+        result = byte_result.decode()
+        result = result. rstrip()
```

Comment:
@TroyLMarshall Ok if you are going for clarity, but not necessary. Could just be: `return byte_result.decode().rstrip().  Are you gaining something from putting it in three steps?

Judgement call, but seems a little more than necessary.